--- src/baseline-lib/geometrize/geometrize/shape/polyline.cpp	2026-08-27 13:24:27.405683800 +0800
+++ src/improved-lib/geometrize/geometrize/shape/polyline.cpp	2026-08-29 12:52:10.548493900 +0800
@@ -1,33 +1,35 @@
-#include "polyline.h"
-
-#include <cstdint>
-#include <memory>
-#include <utility>
-#include <vector>
-
-#include "shape.h"
-
-namespace geometrize
-{
-
-Polyline::Polyline(const std::vector<std::pair<float, float>>& points) : Shape()
-{
-    m_points = points;
-}
-
-std::shared_ptr<geometrize::Shape> Polyline::clone() const
-{
-    std::shared_ptr<geometrize::Polyline> polyline{std::make_shared<geometrize::Polyline>()};
-    polyline->m_points = m_points;
-    polyline->setup = setup;
-    polyline->mutate = mutate;
-    polyline->rasterize = rasterize;
-    return polyline;
-}
-
-geometrize::ShapeTypes Polyline::getType() const
-{
-    return geometrize::ShapeTypes::POLYLINE;
-}
-
-}
+#include "polyline.h"
+
+#include <cstdint>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "shape.h"
+
+namespace geometrize
+{
+
+Polyline::Polyline(const std::vector<std::pair<float, float>>& points) : Shape()
+{
+    m_points = points;
+}
+
+std::shared_ptr<geometrize::Shape> Polyline::clone() const
+{
+    std::shared_ptr<geometrize::Polyline> polyline{std::make_shared<geometrize::Polyline>()};
+    polyline->m_points = m_points;
+    polyline->setup = setup;
+    polyline->mutate = mutate;
+    polyline->mutateScaled = mutateScaled;
+    polyline->rasterize = rasterize;
+    polyline->rasterizeInto = [rf = polyline->rasterize](const geometrize::Shape& s, std::vector<geometrize::Scanline>& out) { out = rf(s); };
+    return polyline;
+}
+
+geometrize::ShapeTypes Polyline::getType() const
+{
+    return geometrize::ShapeTypes::POLYLINE;
+}
+
+}
